### DISCUSSION TITLE
ZF1-Future Issues

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2303] [General]
I am opening this post for discussing issues found in using ZF1-Future framework.  In other post #2194 I created a comparison between OM-20 and ZF1-Future files. What worries me are the custom changes made by the Magento team in /app/code/core/Zend.

For testing use the following: 

- OM-20 latest
- https://github.com/Shardj/zf1-future latest
- change directory names for "/app/code/core/Zend" and "/lib/Zend" to keep them there untouched
- take from zf1-future-release archive this directory /library/Zend and copy it in /lib/Zend

if you encounter problems please post them here with all the details.

**Comment by addison74** [comment]
OM-20.14 + Magento Sample Pack. 

Both Frontend and Backend won't load without these two files /app/code/core/Zend/Db/Select.php and Statement.php. In browser you will get "This page isn't working" HTTP ERROR 500 and there is not line in webserver error log.

Right now in /app/code/core/Zend there is only Db directory. Otherwise the others have changed their name and I did not encounter any issues. Most likely I don't think we will be able to get rid of a ZF1-Future custom version that will be maintained in /app/code/core like Magento team did it before.

**Comment by addison74** [comment]
Let's pay attention to the deprecated section https://php.watch/versions/8.2 for "PHP 8.2: Mbstring: Base64, Uuencode, QPrint, and HTML Entity encodings are deprecated". 

Function **mb_convert_encoding** is used in /lib/Zend and /lib/Pelago. I haven't checked ZF1-Future yet, but they will definitely make the necessary changes. We should think about moving in the direction of ZF1-Future at least for /lib/Zend, then make the custom version in /app/code/core/Zend (without it OM does not work).

For removing deprecated **utf8_encode()** there is a PR here https://github.com/OpenMage/magento-lts/pull/2404.

**Reply by fballiano** [reply]
I've created https://github.com/Shardj/zf1-future/pull/252 for zf1-future

**Reply by fballiano** [reply]
Pelago is already adding support for PHP 8.2 https://github.com/MyIntervals/emogrifier/pull/1163

**Reply by fballiano** [reply]
If I'm not mistaken there are no occurrencies of Base64, Uuencode, QPrint, and HTML Entity encodings:
<img width="672" alt="Schermata 2022-08-11 alle 14 17 54" src="https://user-images.githubusercontent.com/909743/184142445-1246bbf2-0b91-44ee-9238-a65fa68b59d7.png">

**Comment by addison74** [comment]
I used https://github.com/OpenMage/magento-lts/search?q=mb_convert_encoding to get the search results for OpenMage. For the other functions I did not search yet. 

My opinion that the changes to 8.2 do not require great efforts like those when it went from 7 to 8. Maybe there are still undiscovered issues with version 8.1, but if it is not put to work we will never know about them. There in production is the biggest test.

**Comment by addison74** [comment]
Meanwhile ZF-Future became part of OpenMage.
